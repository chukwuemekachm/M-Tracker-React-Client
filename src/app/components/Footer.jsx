import React from 'react';
import '../assets/css/footer.css';

/**
 * @description The Footer component
 *
 * @returns {object}
 */
const Footer = () => (
  <div className="w3-flat-midnight-blue footer">
    Copyright of &copy;2018 Maintenance Tracker by Chima Chukwuemeka
  </div>
);

export default Footer;
