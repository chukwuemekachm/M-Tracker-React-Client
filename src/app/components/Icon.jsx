import React from 'react';
import PropTypes from 'prop-types';
import '../assets/css/icon.css';

/**
 * @description The Icon component
 *
 * @param {string} title The title of the icon
 * @param {object} icon The icon image/svg
 *
 * @returns {object}
 */
const Icon = ({ title, icon }) => (
  <div className="col-md-3 text-center">
    <div className="icon-container">
      <img src={icon} className="about-img" alt="authenticate" />
    </div>
    <h4>
      {title}
    </h4>
  </div>
);

Icon.propTypes = {
  title: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
};

export default Icon;
