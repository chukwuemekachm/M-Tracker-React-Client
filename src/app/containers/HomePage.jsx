import React from 'react';
import { Link } from 'react-router-dom';
import DefaultNavBar from '../components/NavBar';
import Icon from '../components/Icon';
import Footer from '../components/Footer';
import '../assets/css/home.css';

/**
 * @description The Home page component
 *
 * @returns {object}
 */
const HomePage = () => (
  <div>
    <DefaultNavBar />
    <div className="jumbotron">
      <div className="banner">
        <div className="w3-display-middle">
          <h3 className="text-center">
            Facilities don
            {'\''}
            t need to go bad any longer,
            {' '}
            request a repair or maintenance schedule from us today.
          </h3>
          <br />
          <div className="row">
            <div className="col-md-3" />
            <div className="col-md-6">
              <Link to="/signup">
                <button type="button" className="btn btn-success btn-lg btn-block">
                  Get Started
                </button>
              </Link>
            </div>
            <div className="col-md-3" />
          </div>
        </div>
      </div>
    </div>
    <div className="container">
      <br />
      <h3 className="display-6 text-center">
        About Maintenance Tracker
      </h3>
      <br />
      <div className="container">
        <p className="text-center">
          Maintenance Tracker is an platform designed, to provide users with the oppurtunity of
          {' '}
          reaching out to operations or repairs department regarding repair or maintenance
          {' '}
          requests. It also enables users to monitor the status of their requests.
        </p>
      </div>
    </div>
    <br />
    <br />
    <div className="container">
      <h3 className="display-6 text-center">
        How it Works
      </h3>
      <br />
      <br />
      <div className="row w3-display-center">
        <Icon
          title="Authenticate"
          icon="https://res.cloudinary.com/dvcaeuvee/image/upload/v1538265100/undraw_login_jdch.svg"
        />
        <Icon
          title="Log a request"
          icon="https://res.cloudinary.com/dvcaeuvee/image/upload/v1538265099/undraw_add_file2_gvbb.svg"
        />
        <Icon
          title="Request gets approved"
          icon="https://res.cloudinary.com/dvcaeuvee/image/upload/v1538265100/undraw_hiring_cyhs.svg"
        />
        <Icon
          title="Request gets resolved"
          icon="https://res.cloudinary.com/dvcaeuvee/image/upload/v1538265100/undraw_confirmation_2uy0.svg"
        />
      </div>
    </div>
    <Footer />
  </div>
);

export default HomePage;
