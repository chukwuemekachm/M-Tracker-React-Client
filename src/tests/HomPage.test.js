import React from 'react';
import { shallow } from 'enzyme';
import HomePage from '../app/containers/HomePage';
import Footer from '../app/components/Footer';
import Icon from '../app/components/Icon';
import DefaultNavBar from '../app/components/NavBar';

describe('Homepage component', () => {
  it('should render without errors', () => {
    const wrapper = shallow(<HomePage />);
    expect(wrapper.find('p')).toHaveLength(1);
    expect(wrapper.find('h3')).toHaveLength(3);
    expect(wrapper.find('div')).toHaveLength(12);
    expect(wrapper.find(Footer)).toHaveLength(1);
    expect(wrapper.find(Icon)).toHaveLength(4);
    expect(wrapper.find(DefaultNavBar)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
