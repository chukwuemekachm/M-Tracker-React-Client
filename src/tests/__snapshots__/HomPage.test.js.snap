// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage component should render without errors 1`] = `
<div>
  <Connect(NavBar) />
  <div
    className="jumbotron"
  >
    <div
      className="banner"
    >
      <div
        className="w3-display-middle"
      >
        <h3
          className="text-center"
        >
          Facilities don
          '
          t need to go bad any longer,
           
          request a repair or maintenance schedule from us today.
        </h3>
        <br />
        <div
          className="row"
        >
          <div
            className="col-md-3"
          />
          <div
            className="col-md-6"
          >
            <Link
              replace={false}
              to="/signup"
            >
              <button
                className="btn btn-success btn-lg btn-block"
                type="button"
              >
                Get Started
              </button>
            </Link>
          </div>
          <div
            className="col-md-3"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <br />
    <h3
      className="display-6 text-center"
    >
      About Maintenance Tracker
    </h3>
    <br />
    <div
      className="container"
    >
      <p
        className="text-center"
      >
        Maintenance Tracker is an platform designed, to provide users with the oppurtunity of
         
        reaching out to operations or repairs department regarding repair or maintenance
         
        requests. It also enables users to monitor the status of their requests.
      </p>
    </div>
  </div>
  <br />
  <br />
  <div
    className="container"
  >
    <h3
      className="display-6 text-center"
    >
      How it Works
    </h3>
    <br />
    <br />
    <div
      className="row w3-display-center"
    >
      <Icon
        icon="https://res.cloudinary.com/dvcaeuvee/image/upload/v1538265100/undraw_login_jdch.svg"
        title="Authenticate"
      />
      <Icon
        icon="https://res.cloudinary.com/dvcaeuvee/image/upload/v1538265099/undraw_add_file2_gvbb.svg"
        title="Log a request"
      />
      <Icon
        icon="https://res.cloudinary.com/dvcaeuvee/image/upload/v1538265100/undraw_hiring_cyhs.svg"
        title="Request gets approved"
      />
      <Icon
        icon="https://res.cloudinary.com/dvcaeuvee/image/upload/v1538265100/undraw_confirmation_2uy0.svg"
        title="Request gets resolved"
      />
    </div>
  </div>
  <Footer />
</div>
`;
